xlators/performance/open-behind/src/open-behind.c

```diff
@@ -506,7 +506,9 @@ ob_open(call_frame_t *frame, xlator_t *this, loc_t *loc, int flags, fd_t *fd,
 
     LOCK(&fd->inode->lock);
     {
-        ob_inode->open_count--;
+        if (ob_inode != NULL) {
```

Comment:
this is OK, but you have no reason to do the locking in the 1st place if ob_inode is NULL.

Generally, under what conditions are we OK with the code overall, if ob_inode is NULL? I think only error conditions will do that (or state == OB_STATE_READY) - so we can probably add a check before (line 461?) and bail?

Reply:
I would move the check for `ob_inode != NULL` outside the lock, but I would keep it here. The chances that `ob_open_behind()` fails are minimum (basically only a memory error), so we don't need to do checks that most likely will fail. I think it's fine to keep it here, but outside the lock.